openai/openai.go

```diff
@@ -269,10 +271,50 @@ func toModel(r api.ShowResponse, m string) Model {
 	}
 }
 
-func fromChatRequest(r ChatCompletionRequest) api.ChatRequest {
+func fromChatRequest(r ChatCompletionRequest) (api.ChatRequest, error) {
 	var messages []api.Message
 	for _, msg := range r.Messages {
-		messages = append(messages, api.Message{Role: msg.Role, Content: msg.Content})
+		switch content := msg.Content.(type) {
+		case string:
+			messages = append(messages, api.Message{Role: msg.Role, Content: content})
+		case []any:
+			message := api.Message{Role: msg.Role}
+			for _, c := range content {
+				if data, ok := c.(map[string]any); ok {
+					switch data["type"] {
+					case "text":
+						if text, ok := data["text"].(string); ok {
+							message.Content = text
+						} else {
+							return api.ChatRequest{}, fmt.Errorf("invalid message format")
+						}
+					case "image_url":
+						if urlMap, ok := data["image_url"].(map[string]any); ok {
+							if url, ok := urlMap["url"].(string); ok {
+								types := []string{"jpeg", "jpg", "png"}
+								for _, t := range types {
+									url = strings.TrimPrefix(url, "data:image/"+t+";base64,")
+								}
+								if img, err := base64.StdEncoding.DecodeString(url); err == nil {
+									message.Images = append(message.Images, img)
+								}
+							} else {
```

Comment:
This is a big "pyramid" of if statements – is there a way we can flatten it a bit? Generally if else is returning, you can put it before hand and check the opposite and "short circuit" the return